pkg/api/helpers.go

```diff
@@ -38,8 +55,12 @@ var Semantic = conversion.EqualitiesOrDie(
 		if b.Amount == nil && a.MilliValue() == 0 {
 			return true
 		}
+		if a.Amount == nil || b.Amount == nil {
```

Comment:
Note that a & b can't be nil. I made quantities safe to use in an uninitialized state.